src/inputs/BasicInputs/BasicNumberInput/BasicNumberInput.js

```diff
@@ -17,37 +17,63 @@ export const BasicNumberInput = (props) => {
   const { id, label, tooltipText, value, textFieldProps, inputProps, changeNumberField, isDirty, ...otherProps } =
     props;
 
-  const [textInput, setTextInput] = useState(value.toString());
+  const convToStringValue = useCallback((numValue) => {
+    if (numValue === undefined || numValue === null) {
+      return '';
+    } else {
+      return numValue.toString();
+    }
+  }, []);
+
+  const [textInput, setTextInput] = useState(convToStringValue(value));
+
+  const setTextInputByValue = useCallback(() => {
+    const stringValue = convToStringValue(value);
+    if (textInput !== stringValue) {
+      setTextInput(stringValue);
+    }
+  }, [convToStringValue, textInput, value]);
+
+  const isFocused = useRef(false);
   useEffect(() => {
-    if (textInput && parseFloat(textInput) !== value) setTextInput(value.toString());
-  }, [value, textInput]);
+    if (!isFocused.current) setTextInputByValue();
+  }, [setTextInputByValue]);
+
+  const handleFocusEvent = useCallback(() => {
+    isFocused.current = true;
+  }, []);
 
   const handleChangeEvent = useCallback(
     (event) => {
-      setTextInput(event.target.value);
-      const inputValueAsNumber = parseFloat(event.target.value);
-      if (inputValueAsNumber !== value) changeNumberField(inputValueAsNumber);
+      const textValue = event.target.value;
+      const numValue = parseFloat(textValue);
+      if (!isNaN(numValue)) {
+        changeNumberField(parseFloat(numValue));
```

Comment:
I think `parseFloat(numValue)` should be only `numValue`, we don't need to parse it again :slightly_smiling_face: